app/assets/javascripts/topology_preview.js

```diff
@@ -1,33 +1,34 @@
 var TopologyPreviewer = (function(){
-  var topologyGraph, svgGroup, data, root, tree, svgHeight, svgWidth;
+  var topologyGraph, presetData, style, svgGroup, data, root, tree, svgHeight, svgWidth,
+      currentStylesheet;
 
   var nodeIds         = 0,
       maxLabelLength  = 20,
-      duration        = 250,
-      endPointCounter = 0,
-      stepSize        = 20,
       ease            = 'cubic-out';
 
   var diagonal = d3.svg.diagonal()
-      .projection(function(d) { return [d.y, d.x]; });
+      .projection(function(d) { return [d.x, d.y]; });
 
   TopologyPreviewer.prototype = {
     preview: function(){
       clear();
 
-      svgHeight = viewerHeight();
-      svgWidth  = viewerWidth();
-      svg       = buildBaseSvg();
-      data      = [graphData()];
-      root      = data[0];
+      currentStylesheet = TopologyStylesheet[style];
```

Comment:
> overwrite the vars of the first? Putting them inside a closure means they're not globally readable, but still only one can exist at any moment in time, right? Maybe I'm misunderstanding?

Yes. That's what happens. This is mainly due to the fact that there's only one previewer actually doing the `previewing`. But I'll change it to the `this.`-notation. To give each new object their own set of variables.

I'm using a bit of a mash-up from two design patterns which was once advised to me in a fancy blog article which I can't find anymore.

```javascript
var Car = (function(){ 
  Car.prototype = { 
     public: method(){ 
       return 5;
     } 
  };
   
  function private(){
  };
   
  function Car(){
  };
  
  return Car;
})();
````

Also, I think `Coffeescript` 'converts' it in such a fashion.